driver-core/src/main/java/com/datastax/driver/core/CodecRegistry.java

```diff
@@ -0,0 +1,760 @@
+
+/**
+ * A registry for {@link TypeCodec}s. When the driver
+ * needs to serialize or deserialize an object,
+ * it will lookup in the {@link CodecRegistry} for
+ * a suitable codec.
+ *
+ * <h3>Usage</h3>
+ *
+ * <p>
+ * To build a {@link CodecRegistry} instance with all the default codecs used
+ * by the Java driver, simply use the following:
+ *
+ * <pre>
+ * CodecRegistry myCodecRegistry = new CodecRegistry().registerDefaults();
+ * </pre>
+ *
+ * Custom {@link TypeCodec}s can be added to a {@link CodecRegistry} via the {@code register} methods:
+ *
+ * <pre>
+ * CodecRegistry myCodecRegistry = new CodecRegistry()
+ *    .register(myCodec1, myCodec2, myCodec3)
+ *    .registerDefaults(); // to ensure that all CQL types are covered by a matching codec
+ * </pre>
+ *
+ * To be used by the driver, {@link CodecRegistry} instances must then
+ * be associated with a {@link Cluster} instance:
+ *
+ * <pre>
+ * CodecRegistry myCodecRegistry = ...
+ * Cluster cluster = new Cluster.builder().withCodecRegistry(myCodecRegistry).build();
+ * </pre>
+ *
+ * To retrieve the {@link CodecRegistry} instance associated with a Cluster, do the
+ * following:
+ *
+ * <pre>
+ * CodecRegistry registry = cluster.getConfiguration().getCodecRegistry();
+ * </pre>
+ *
+ * Registered codecs will remain in a reservoir where they can
+ * potentially be selected to perform serialization or
+ * deserialization operations requested by the application.
+ * <p>
+ * Note that the order in which codecs are added to the registry matters;
+ * any codec that is registered may potentially override an already registered one.
+ * When looking for a matching codec, the registry will consider all known codecs
+ * starting with the last ones and will return the first matching candidate,
+ * even if another codec would be a better match.
+ * <p>
+ * For most users, the default {@link CodecRegistry} instance should be adequate.
+ * <strong>Using customized codecs is an advanced feature and registries
+ * that do not have enough codecs or codecs that do not comply with the native protocol
+ * specifications may cause the driver to crash.</strong>
+ *
+ * <h3>Examples</h3>
+ *
+ * <p>
+ * <strong>1. Globally use a custom codec to handle a given CQL type</strong>
+ *
+ * <p>
+ * E.g. let's suppose you want to have all your CQL timestamps
+ * deserialized as {@code java.time.DateTime} instances:
+ *
+ * <pre>
+ * TypeCodec&lt;DateTime> timestampCodec = ...
+ * new CodecRegistry().register(timestampCodec).registerDefaults();
+ * </pre>
+ *
+ * <p>
+ * <strong>2. Use a special codec for a specific keyspace, table and column</strong>
+ *
+ * <p>
+ * E.g. let's suppose you want to have a certain varchar column deserialized as a JSON object:
+ *
+ * <pre>
+ * TypeCodec&lt;MyJsonObject> jsonCodec = ...
+ * new CodecRegistry()
+ *      .register(jsonCodec, "myKeyspace.myTable.myColumn") // will be used only for this specific column
+ *      .registerDefaults();
```

Comment:
`registerDefaults` does not exist, instead a new registry gets the default codecs by default.